src/dispatch.c

```diff
@@ -48,6 +49,35 @@ SEXP method_internal(SEXP table, SEXP signature, R_xlen_t signature_itr, SEXP ig
   return R_NilValue;
 }
 
+void R7_method_lookup_error(SEXP generic, SEXP signature) {
+  static SEXP R7_method_lookup_error_fun = NULL;
+  SEXP ns = Rf_findVarInFrame(R_NamespaceRegistry, Rf_install("R7"));
+
+  if (R7_method_lookup_error_fun == NULL) {
+    R7_method_lookup_error_fun = Rf_findVarInFrame(ns, Rf_install("method_lookup_error"));
+  }
+  SEXP name = Rf_getAttrib(generic, Rf_install("name"));
+  SEXP args = Rf_getAttrib(generic, Rf_install("dispatch_args"));
+  SEXP R7_method_lookup_error_call = PROTECT(Rf_lang4(R7_method_lookup_error_fun, name, args, signature));
+  Rf_eval(R7_method_lookup_error_call, ns);
+}
```

Comment:
I'm not sure if this will yell at you or not if you run it through rchk. I know we do similar things in rlang/vctrs but I think we also tag those functions with `__attribute__ ((noreturn))`

https://github.com/r-lib/rlang/blob/6e9ef5ee1aaab20dc9d2a04b7a3be043e869f4fc/src/rlang/rlang-types.h#L21
https://github.com/r-lib/rlang/blob/570dde18f2aa3a5817206acecbe9c56d4dddbea0/src/rlang/cnd.c#L73